pkg/cmd/util/cmd.go

```diff
@@ -31,3 +33,32 @@ func GetDisplayFilename(filename string) string {
 
 	return filename
 }
+
+// ResolveResourceSyntax will return the kind (if found), name and a possible error
+// from the provided arg. If a default kind has been specified, then arg
+// should respect it.
+func ResolveResourceSyntax(mapper meta.RESTMapper, defaultKind, arg string) (string, string, error) {
+	if mapper == nil {
+		return "", "", errors.New("mapper cannot be nil")
+	}
+
+	var name, kind string
+	parts := strings.Split(arg, "/")
+	switch len(parts) {
+	case 1:
+		name = parts[0]
+	case 2:
+		_, kind, err := mapper.VersionAndKindForResource(parts[0])
+		if err != nil {
+			return "", "", err
+		}
+		if len(defaultKind) > 0 && kind != defaultKind {
```

Comment:
I've renamed to `requiredKind` because that's what its' supposed to be. This function is supposed to be called instead of the resource builder when we don't want to fetch the object but it shouldn't default to a resource, and allow others as well (how `ResourceNames` works), it should enforce the resource if it's specified (if not, then we just preserve `oc cmd NAME` syntax).